java-core/pom.xml

```diff
@@ -39,10 +39,11 @@
       </dependency>
       <!-- Test dependencies -->
       <dependency>
-        <groupId>junit</groupId>
-        <artifactId>junit</artifactId>
-        <version>4.13.2</version>
-        <scope>test</scope>
+        <groupId>org.junit</groupId>
+        <artifactId>junit-bom</artifactId>
+        <version>5.10.2</version>
+        <type>pom</type>
+        <scope>import</scope>
```

Comment:
Might be missing some context about dependency management with java-core and/or test jars, but I would have assumed this version would be been defined in pom-parent or shared-deps instead of having to declare it here.

Reply:
I think declare a test dependency in java-core is fine since it allows other modules to use different testing framework.

That being said, we can declare test dependencies in a centralized place, e.g., third-party dependency. However, this should be a separate task.

Reply:
@suztomo Do you have any suggestions of what's the best way to manage junit 5 dependencies?

Reply:
Avoid declaring junit bom in the dependencyManagement section of pom.xml files that would go to the Libraries BOM, such as the pom-parent or gapic-generator-java-bom. That would control library users' junit usage.

Third-party-dependencies is good but java-core, I believe, does not import it.

How about declaring the JUnit version as a property, not dependencyManagement section, in the pom-parent?

Reply:
@suztomo thanks for the suggestion.